src/pint/toa.py

```diff
@@ -1417,6 +1424,59 @@ def compute_posvels(self, ephem=None, planets=False):
         log.debug("Adding columns " + " ".join([cc.name for cc in cols_to_add]))
         self.table.add_columns(cols_to_add)
 
+    def add_vel_ecl(self):
```

Comment:
This function needs an input of the ECL angle.

Reply:
Can you clarify?  Isn't that handled by the PulsarEcliptic class?

Reply:
Why does the function need the ECL angle?

Reply:
The ECL angle value is specified by the .par file. The PulsarEcliptic class has a default value, but it can be changed when initializing it. See code:
 https://github.com/nanograv/PINT/blob/3c33ad5f46fc4928cb42b98449475d324b40f367/src/pint/models/astrometry.py#L475-L503

Reply:
Here the class PulsarEcliptic is used in `.transform_to(PulsarEcliptic)`.  So I guess it needs to be `.transform_to(PulsarEcliptic(obliquity=obliquity))` as in L555 of astrometry.py?
But, that means that this TOAs table column will have a dependency on that setting, in the same way that the TDB column depends on the EPHEM setting.  So probably that setting needs to be saved in the TOAs object the same was as ts.ephem holds the value of ephem used when computing the table columns.

Reply:
yes, if the Vel_ecl column is added to the table, we should record it. When applying toas to a timing model under the ecliptic coordinate, the value in the table should be compared with the value in the timing model.

Reply:
So I need to extract 'obliquity' to store in the TOAs object, then apply it with `.transform_to(PulsarEcliptic(obiquity=self.obliquity))`?

Reply:
Yep, that seems right. See L505 of astrometry.py for how to grab the obliquity value. I don't think you want the `self.obliquity` in your line above.  The local var `obliquity` is set based on `self.ECL.value`which is a parameter of the timing model.  I assume that every model with an `AstrometryEcliptic` component has an ECL set, right @luojing1211 ?

Reply:
Looks like @champagne-cmd has fixed that up now. Thanks for catching that @luojing1211 !